MH-12879, Default location of paella configuration

lkiesow opened the pull request:
The paella player lacks any default configuration for the location of
its main configuration files and is thus failing if this key is not
explicitly set, even if the default configuration file is provided.

This patch adds a default to `${karaf.etc}/paella` which is already
being used for the default configuration file.

lkiesow committed:
MH-12879, Default location of paella configuration

The paella player lacks any default configuration for the location of
its main configuration files and is thus failing if this key is not
explicitly set, even if the default configuration file is provided.

This patch adds a default to `${karaf.etc}/paella` which is already
being used for the default configuration file.

staubesv commented on etc/custom.properties:
```diff
@@ -146,7 +146,10 @@ org.opencastproject.db.ddl.generation=true
 
-org.opencastproject.engage.paella.config.folder=${karaf.etc}/paella
+# Path to the Paella player configuration files. Note that the configuration file for each tenant needs to be in a
+# subfolder with the tenants identifier, e.g. …/etc/mh_default_org/config.json
```
Hmm... is that multi-tenant stuff really true? I can't see the logic that locates the file in a subfolder <tenant identifier> of 'etc' ....

lkiesow replied:
[I can](https://github.com/lkiesow/opencast/blob/46c6194fc84e423cb22dfe15f054e42593f9e295/modules/engage-paella-player/src/main/java/org/opencastproject/engage/paella/PaellaConfigRest.java#L113) ;-P

staubesv replied:
Oh, should have had a closer look to the code... thanks ;-)

staubesv commented on modules/engage-paella-player/src/main/java/org/opencastproject/engage/paella/PaellaConfigRest.java:
```diff
@@ -79,9 +83,16 @@ protected void setSecurityService(SecurityService securityService) {
+      if (StringUtils.isBlank(paellaConfigFolder)) {
+        throw new ConfigurationException("Paella configuration not set and unable to fall back to default location");
+      }
+      paellaConfigFolder += "/paella";
```
Hmm... Using a platform-independent method that uses the correct path separator would be preferable, wouldn't it?

lkiesow replied:
This supports all platforms Opencast supports. But I can change the concatenation method if you want.

staubesv replied:
Let's change it for sake of cleaner code ;-)

lkiesow replied:
Interesting. For me, this is the simpler and cleaner variant. The alternative does make the code more complex. But will push a patch in a minute.

lkiesow committed:
MH-12879, Platform independent path concatenation

This patch uses a file constructor instead of a hard-coded slash for
path separation.
